src/ice_pond.F90

```diff
@@ -0,0 +1,513 @@
+use SIS_types,         only : ice_state_type, ist_chksum
+use fms_io_mod,        only : register_restart_field, restart_file_type
+use SIS_tracer_registry, only : SIS_tracer_registry_type, SIS_tracer_type, get_SIS_tracer_pointer
+use SIS2_ice_thm,      only : get_SIS2_thermo_coefs
+use ice_grid,          only : ice_grid_type
+!Icepack modules
+use icepack_kinds
+use icepack_itd, only: icepack_init_itd, cleanup_itd
+use icepack_meltpond_cesm, only: compute_ponds_cesm
+use icepack_meltpond_lvl,  only: compute_ponds_lvl
+use icepack_meltpond_topo, only: compute_ponds_topo
+use icepack_warnings, only: icepack_warnings_flush, icepack_warnings_aborted, &
+                            icepack_warnings_setabort
+use icepack_tracers, only: icepack_init_tracer_indices, icepack_init_tracer_sizes
+use icepack_tracers, only: icepack_query_tracer_sizes
+use icepack_parameters, only : icepack_init_parameters
+implicit none ; private
+
+#include <SIS2_memory.h>
+
+public :: ice_ponds, ice_ponds_init
+
+type, public :: ice_ponds_CS ; private
+  logical :: &
+  cesm_pond = .false., &       !< .true. = deprecated CESM ponds
+  level_pond = .false., &      !< .true. = preferred ponds on level ice
+  topo_pond = .false.          !< .true. = topographic ponds
+  real :: area_underflow = 0.0 !< a non-dimesional fractional area underflow limit for the sea-ice
+                               !! ridging scheme. This is defaulted to zero, but a reasonable
+                               !! value might be 10^-26 which for a km square grid cell
+                               !! would equate to an Angstrom scale ice patch.
+end type ice_ponds_CS
+
+contains
+
+subroutine ice_ponds_init(G, IG, PF, CS, US)
+  type(SIS_hor_grid_type),    intent(in) :: G      !<  G The ocean's grid structure.
+  type(ice_grid_type),        intent(in) :: IG     !<   The sea-ice-specific grid structure.
+  type(param_file_type),      intent(in) :: PF     !< A structure to parse for run-time parameters
+  type(ice_ponds_CS),       pointer    :: CS     !< The ponds control structure.
+  type(unit_scale_type),      intent(in) :: US     !< A structure with unit conversion factors.
+
+  integer (kind=int_kind) :: ntrcr, ncat, nilyr, nslyr, nblyr, nfsd, n_iso, n_aero
+  integer (kind=int_kind) :: nt_Tsfc, nt_sice, nt_qice, nt_alvl, nt_vlvl, nt_qsno
+  character(len=40) :: mdl = "ice_ponds_init" ! This module's name.
+
+  if (.not.associated(CS)) allocate(CS)
+! call get_param(PF, mdl, "NEW_RIDGE_PARTICIPATION", CS%new_rdg_partic, &
+!                "Participation function used in ponds, .false. for Thorndike et al. 1975 "//&
+!                ".true. for Lipscomb et al. 2007", default=.false.)
+! call get_param(PF, mdl, "NEW_RIDGE_REDISTRIBUTION", CS%new_rdg_redist, &
+!                "Redistribution function used in ponds, .false. for Hibler 1980 "//&
+!                ".true. for Lipscomb et al. 2007", default=.false.)
+
+  ncat=IG%CatIce ! The number of sea-ice thickness categories
```

Comment:
Please use white space around the equal signs here, as in `nilyr = IG%NkIce`.  As discussed in the MOM6 style guide at https://github.com/NOAA-GFDL/MOM6/wiki/Code-style-guide, which also pertains to SIS2, we reserve the use of the equal sign without spaces for named arguments, as in `default=.false.` two lines above.  This is a minor stylistic thing, but it does help when searching through the code for irregular patterns that might be bugs, and we think that it promotes the human readability of the code.